### ISSUE TITLE
check_calibconstr has "else if" where "if" should be used.

### ISSUE THREAD

**Issue by extrospective** [facebookexperimental/Robyn#275]
## Describe issue
Small code improvement to be done by changing "else if" to "if" and thereby ensuring a branch of code is not accidentally skipped with a significant warning.

This is an insignificant change and in itself may have little value other than making code more precisely correct.

From code inspection.
This is from the code at github as of 1/25/2022.
Line 452 of checks.R

Code today reads:
```check_calibconstr <- function(calibration_constraint, iterations, trials, calibration_input) {
  if (!is.null(calibration_input)) {
    total_iters <- iterations * trials
    if (calibration_constraint < 0.01 | calibration_constraint > 0.1) {
      calibration_constraint <- 0.1
      message("calibration_constraint must be >=0.01 and <=0.1. Using default value 0.1")
    } else if (total_iters * calibration_constraint < 500) {
      warning("Calibration constraint set to be top ", calibration_constraint*100, "% calibrated models.",
              " Only ", round(total_iters*calibration_constraint,0), " models left for pareto-optimal selection")
    }
  }
  return(calibration_constraint)
}
```

If the test fails where calibration constraint is not in 0.01 to 0.1 range it is reset to 0.1.
But then the next branch of code does not run.

Even if the first test fails it seems that the warning should still fire when the total_iters and calibration_constraint (now 0.1) is insufficient to generate  500 models.

## Suggestion for one enhancement while touching these lines of code

While the code is being changed, I might recommend that the 500 "magic number" be changed to a named constant for comparison and that this named constant be compared with and also output in the warning so the user does not have to read the code to learn the constraint which is being compared with.  

I suppose it is stylistic to prefer named constants over magic numbers; it was a coding recommendation years ago and there are pros and cons of course.

## Commentary

Not sure why these constraints of 0.01 and 0.1 were determined.  Especially the 0.1 constraint seems to me that it could be relaxed to 1.0, unless you think such users are truly trying to shoot themselves in the foot.

As my use of Robyn explores calibration, I am thinking a use is a quasi-experiment where one gently guides nevergrad and model selection to more plausible models. These quasi-experiments would also be necessary if one did the experiment outside the date window, and yet wanted to inform the models in the date window.  But when doing a quasi-experiment, maybe one does not want to be as stringent as 0.1 (10% of models).  My conclusion is that I might come back at a future date and propose allowing values greater than 0.1.

**Comment by laresbernardo** [comment]
Thanks for the feedback and catching that @extrospective 
Will commit code change directly to main branch in a moment.
Re 10% minimum forced: We think users should just run more iterations if they want to have more samples at the end. It means that if we run 10k iterations, only the top 1k will be considered for pareto output among NRMSE & RSSD. If we want more samples for pareto, we can increase to 20k etc, but we wouldn't want to loosen calibration constraints even more.

**Comment by gufengzhou** [comment]
Bit more context: The main reason is that we believe causal experiments are source of truth. When users actively decide to calibrate the model, we want to force the calibration error (MAPE.LIFT) to be prioritized. This is a believe-based decision from us.

**Comment by extrospective** [comment]
One comment on the 10%...

I think a point we can discuss (perhaps in another channel) is how the model responds to calibration "truth".  My sense from testing so far is that if I provide a single day of calibration experience on a single channel, that one piece of "truth" will exert an incredible effect on the overall model.   (This experiment is a boundary condition for the calibration system.)

So I could imagine that 10% might be reasonable if you had a fair amount of calibration data to take into account, but that users might determine how much "weight" to give to the calibration observations.  Maybe our intuition is that 10% would apply when half of the channels have calibration data for 10% of the observations for example.

In short, the 10% default may seem fine, but I question code overruling the user and saying the top 10 percentile is the **most** tolerant one could be of any calibration.

**Comment by gufengzhou** [comment]
I understand what you mean. As mentioned before, this decision is believe-based. Just as your suggestion of "10% quantile for 10% calibrated data points", it's an arbitrary decision. 

If we agree that causality is the closest we can get to truth, then we don't really care about how strong it manipulates the model. The manipulation is rather desired. Even if we only have 1 day of experimental result, we believe its better than 0 day. 

In the end, users are free to modify Robyn because it's open source. We do our best to be transparent about our decisions and be open for suggestions. But we also stand by principals we firmly believe in and hope our reasoning behind is comprehensible to users.

**Comment by laresbernardo** [comment]
Closing this ticket now. Let us know if you need further help or have any related comments
